sparse_list.py

```diff
@@ -144,13 +147,18 @@ def __initialise_from_iterable(self, arg):
         self.size = len(self.elements)
 
     def __eq__(self, other):
-        return all(a == b for a, b in zip_longest(self, other))
+        return len(self) == len(other) and all(a == b for a, b in zip_longest(self, other))
```

Comment:
Having asserted that the two sequences are the same length, we can remove the call to `zip_longest()`

```suggestion
        return len(self) == len(other) and all(a == b for a, b in zip(self, other))
```

Reply:
And this means we can remove `from six.moves import zip_longest`